### PR TITLE
Read cache - fio load test script

### PR DESCRIPTION
**Pull request by raj-prince**
### Description
Adding script to run the fio based workload for read cache specific workload.
1. Support to run epoch based workload using fio.
2. Cleaning the kernel cache after every epoch.

#### How to run the script?
Running a particular workload contains two major steps: 
(a) Workload directory - it can be a gcsfuse mounted directory or any nomal directory (local-ssd). Here, mounting of gcsfuse will be done by `perfmetrics/scripts/read_cache/mount_gcsfuse.sh`. Please run with `-h` flag to get the available options.
(b) Running the fio load test for the given workload directory. This will be done by `perfmetrics/scripts/read_cache/run_read_cache_fio_workload.sh`. Please run with -h flag to know the available options.

E.g.

``` bash
#!/bin/bash
set -x
set -e
if [[ -z "${WORKING_DIR}" ]]; then
  echo "Please set the working directory..."
  exit 1
fi
num_of_threads=${1:-40}
ONE_MILLION=2000
FIFTY_K=50000
cd $WORKING_DIR/gcsfuse/perfmetrics/scripts/read_cache/
# For file-size = 1K and block-size = 1K
./mount_gcsfuse.sh -s 1 -c /mnt/disks/local_ssd -b gcsfuse-read-cache-fio-load-test
workload_dir=$WORKING_DIR/gcs/1K
mkdir -p $workload_dir
number_of_files_per_thread=$((ONE_MILLION / num_of_threads))
./run_read_cache_fio_workload.sh -e 5 -n $number_of_files_per_thread -t $num_of_threads -b 1K -s 1K -d $workload_dir
cd -
```
Here, -s denotes the max_size_in_mb, -c cache_location, -b bucket_name flag used in the mount_gcsfuse.sh command.
And -e <number of epoch>, -n <number of files per thread>, -t <number of threads>, -b <block_size> -s <file_size> and -d is the workload directory.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
4. Unit tests - NA
5. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/read_cache/one_time_setup.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Add go in the path permanently, so that $HOME/go/bin is visible.
+export PATH=$PATH:$HOME/go/bin/
+echo 'export PATH=$PATH:$HOME/go/bin/:/usr/local/go/bin' >> ~/.bashrc
+
+# Export WORKING_DIR env variable and add it to ~/.bashrc.
+export WORKING_DIR=$WD
+echo "export WORKING_DIR=$WD" >> ~/.bashrc
+
+# Install gcsfuse.
+go install github.com/googlecloudplatform/gcsfuse@read_cache_release
```
[the remark this answers is not in the document]
Comment (raj-prince) [reply perfmetrics/scripts/read_cache/one_time_setup.sh:43]: Seems working with go install too.

File: perfmetrics/scripts/read_cache/run_load_test.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+set -x
+
+set -e
+
+if [[ -z "${WORKING_DIR}" ]]; then
+  echo "Please set the working directory..."
+  exit 1
+fi
+
+num_of_threads=${1:-40}
```
Comment (sethiay) [inline perfmetrics/scripts/read_cache/run_load_test.sh:26]: can we make num of threads, number of files passable ? Basis on these two we can decide max file size , number of files per thread.

Also, if it takes time, we are fine to not check-in this script for now and check-in once we run initial tests and are sure about the workloads that we will run repeatedly. 

For now, you can just document the below snippet somewhere. 
./mount_gcsfuse.sh -s 950 -c /mnt/disks/local_ssd
workload_dir=$WORKING_DIR/gcs/1K
mkdir -p $workload_dir
number_of_files_per_thread=$((ONE_MILLION / num_of_threads))
./run_read_cache_fio_workload.sh -e 5 -n $number_of_files_per_thread -t $num_of_threads -b 1K -s 1K -d $workload_dir

File: perfmetrics/scripts/read_cache/cleanup_kernel_cache.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (raj-prince) [reply perfmetrics/scripts/read_cache/cleanup_kernel_cache.sh:1]: Not required now, as deleted this.

File: perfmetrics/scripts/read_cache/run_load_test.sh (hunk 2 above)
Comment (raj-prince) [reply perfmetrics/scripts/read_cache/run_load_test.sh]: Agreed. Removing this file for now.

File: perfmetrics/scripts/read_cache/setup.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+sudo rm -rf "$FIO_SRC_DIR" && \
+git clone https://github.com/axboe/fio.git "$FIO_SRC_DIR" && \
+cd  "$FIO_SRC_DIR" && \
+git checkout fio-3.36 && \
+sed -i 's/define \+FIO_IO_U_PLAT_GROUP_NR \+\([0-9]\+\)/define FIO_IO_U_PLAT_GROUP_NR 32/g' stat.h && \
+./configure && make && sudo make install
+cd -
+
+# Install and validate go.
+version=1.21.1
```
Comment (sethiay) [inline perfmetrics/scripts/read_cache/setup.sh:79]: please change to 1.21.3

**Review by sethiay — COMMENTED** [review]
Please add an example of how to run the script for a file size.

**Comment by raj-prince** [comment]
> Please add an example of how to run the script for a file size.

Updated the description with the example.

**Review by sethiay — APPROVED** [review]

**Pull request merged**